sdk/eventhub/azure-eventhubs/azure/eventhub/aio/receiver_async.py

```diff
@@ -156,18 +165,46 @@ def __aiter__(self):
             self._handler = ReceiveClientAsync(
                 source,
                 auth=self.client.get_auth(**alt_creds),
-                debug=self.client.debug,
+                debug=self.client.config.network_tracing,
                 prefetch=self.prefetch,
                 link_properties=self.properties,
                 timeout=self.timeout,
                 error_policy=self.retry_policy,
                 keep_alive_interval=self.keep_alive,
                 client_name=self.name,
-                properties=self.client.create_properties(),
+                properties=self.client.create_properties(self.client.config.user_agent),
                 loop=self.loop)
-        await self._handler.open_async()
-        while not await self._handler.client_ready_async():
-            await asyncio.sleep(0.05)
+
+        if not self.running:
+            try:
+                await self._handler.open_async()
+                self.running = True
+                while not await self._handler.client_ready_async():
+                    await asyncio.sleep(0.05)
+            except errors.AuthenticationException:
+                    log.info("Receiver failed authentication. Retrying...")
+                    await self.reconnect()
+            except (errors.LinkDetach, errors.ConnectionClose) as shutdown:
+                if shutdown.action.retry and self.auto_reconnect:
+                    log.info("Receiver detached. Attempting reconnect.")
+                    await self.reconnect()
+                else:
+                    log.info("Receiver detached. Failed to connect")
+                    error = ConnectError(str(shutdown), shutdown)
+                    raise error
+            except errors.AMQPConnectionError as shutdown:
+                if str(shutdown).startswith("Unable to open authentication session") and self.auto_reconnect:
+                    log.info("Receiver couldn't authenticate.", shutdown)
```

Comment:
This format string looks incorrect...

Reply:
Yes this is a bug. Changed to 
log.info("Receiver couldn't authenticate (%r).", shutdown)